We Fly Digital

TEMESGEN-G25 opened the pull request:
## Summary

#### Basics
- You should follow the layout of [the provided design](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign). You should change only the titles, descriptions, and photos - in order to create a website about something other than motorcycles.
- Select a theme for your website - is it going to be a website for booking doctor appointments, booking online classes, or something else?

#### Features

**Core features - for teams of each size**
- The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
- In the navigation panel, the user can see links to:
    - Motorcycles/doctors/classes/items that you selected as a theme.
    - "Reserve" form.
    - "My reservations".
    - "Add motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
    - "Delete motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
- On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).

- When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).
    - In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

- Make the app responsive, creating both mobile and desktop versions.


**If your team has 3-4 members these are the required features; otherwise, consider them as optional features.**
- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
    - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
    - Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
    - The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.
- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).
- Add full documentation for your API.


**If your team has 5 members these are the required features; otherwise, consider them as optional features.**

- Implement proper user authentication from the front-end to the server.
- Make sure that the "Add item" and "Delete item" links are accessible only by users who are admins.

#### Technical set up
- Set up the repository/repositories on GitHub and use Gitflow.
- You should use Postgres as your database
- Use Rails to create backend API.
- Use React & Redux to create frontend UI.
- You can choose if you want to set up your project as two separate apps or as one.
- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

ibr5500 committed:
Initialize A React app

ibr5500 committed:
Add React-Redux linters configurations

ibr5500 committed:
Merge pull request #1 from ibr5500/setup

Flight Reservation Front-end: Setup

TEMESGEN-G25 committed:
Remove needless files & create reservation

TEMESGEN-G25 committed:
Install react-router-dom & create reservationForm

TEMESGEN-G25 committed:
Fetch users from the api

TEMESGEN-G25 committed:
Install redux & add redux functionality to users

TEMESGEN-G25 committed:
Display a user from the api

TEMESGEN-G25 committed:
Display each user at the reservation component

KanoCode committed:
Add Component NavigationPanel, add logo

KanoCode committed:
remove logo

KanoCode committed:
Add fontawesome cdn, add styles for navigation panel

KanoCode committed:
Add react-router-dom, set up routes

KanoCode committed:
Add styling to Navigation panel, make responsive

KanoCode committed:
Add sample components, test navigation panel

KanoCode committed:
Fix linters

ibr5500 committed:
Merge pull request #2 from ibr5500/navigation-panel

Navigation panel

ibr5500 committed:
Implement the home page

ibr5500 committed:
Make some edits

TEMESGEN-G25 committed:
Create register & login redux files

TEMESGEN-G25 committed:
Create a form to register a user

ibr5500 committed:
Implement the slider in home page

TEMESGEN-G25 committed:
Install redux toolkit

TEMESGEN-G25 committed:
Add createSlice & use configureStore reduxtoolkit

ibr5500 committed:
Fix url for the flights

GOlivierNation committed:
Merge pull request #4 from ibr5500/home-page

Flight Reservation Fornt-End: Home page

TEMESGEN-G25 committed:
Add createSlice & create builder cases

TEMESGEN-G25 committed:
Display all users from rails api using toolkit

TEMESGEN-G25 committed:
Rename login to register & setup registerSlice

TEMESGEN-G25 committed:
Create new user using registerSlice

TEMESGEN-G25 committed:
Create & setup loginSlice for user login

TEMESGEN-G25 committed:
Merge branch 'dev' into user-sign-up-page

TEMESGEN-G25 committed:
Comment all normal redux related code

TEMESGEN-G25 committed:
Move all components into one Components folder

TEMESGEN-G25 committed:
Add a signUp page

TEMESGEN-G25 committed:
Register a new user using registerSlice

TEMESGEN-G25 committed:
Use react hooks to register user

ibr5500 committed:
Add functionalty for login

ibr5500 committed:
Fetch data from the api

ibr5500 committed:
Fix errors

TEMESGEN-G25 committed:
Add airline route into App.js file

TEMESGEN-G25 committed:
Create a new component AirlineForm

TEMESGEN-G25 committed:
Delete userSlice file & from all other files

TEMESGEN-G25 committed:
Make the fetch user signUp & logIn in one slice

TEMESGEN-G25 committed:
Delete reservationForm

TEMESGEN-G25 committed:
Rename detail folder in redux into airlines folder

ibr5500 committed:
Merge airline-details branch

KanoCode committed:
Update styles , add background image

KanoCode committed:
Update login component

KanoCode committed:
Update signup component

KanoCode committed:
fix linters

KanoCode committed:
Update app.css

TEMESGEN-G25 committed:
Remove all logs

TEMESGEN-G25 committed:
Redirect the logged in user to the homepage

TEMESGEN-G25 committed:
Store the log in token to create new airlines

KanoCode committed:
update App.css

ibr5500 committed:
Fix some errors

ibr5500 committed:
Merge branch 'airline-details' of https://github.com/ibr5500/flight_reservation-frontend into airline-details

TEMESGEN-G25 committed:
Merge branch 'user-sign-up-page' into airline-details

ibr5500 committed:
Created Details page to display a single flight data

ibr5500 committed:
Fix some errors

ibr5500 committed:
Delete unsued code

ibr5500 committed:
Add button tag for details page

TEMESGEN-G25 committed:
Merge branch 'airline-details' of github.com:ibr5500/flight_reservation-frontend into airline-details

TEMESGEN-G25 committed:
Change the login link to button at the login page

ibr5500 committed:
Fixing Carousel

TEMESGEN-G25 committed:
Change the route name from flight to airline

TEMESGEN-G25 committed:
Rename the detail component to AirlineDetail

TEMESGEN-G25 committed:
Create a reservationForm component

ibr5500 committed:
Fix linter errors

ibr5500 committed:
Fix a linter setup error

KanoCode committed:
Add styles for details component

ibr5500 committed:
Fix linter errors

KanoCode committed:
Merge branch 'airline-details' of https://github.com/ibr5500/flight_reservation-frontend into airline-details

KanoCode committed:
Update styles

KanoCode committed:
update styles

KanoCode committed:
update styles

ibr5500 committed:
Fix some linter errors

ibr5500 committed:
Merge pull request #5 from ibr5500/user-sign-up-page

User sign up page

ibr5500 committed:
Fix a confliect

ibr5500 committed:
Fix linter errors

ibr5500 committed:
Add route for reservation

ibr5500 committed:
Sort the routes

ibr5500 committed:
Add reservation reducer

KanoCode committed:
create link to reservation form

TEMESGEN-G25 committed:
Create a slice for fetching & create reservations

KanoCode committed:
Add handlers and input elements to reservationform

KanoCode committed:
Merge branch 'new-reservations' of https://github.com/ibr5500/flight_reservation-frontend into new-reservations

TEMESGEN-G25 committed:
Fix linters

TEMESGEN-G25 committed:
Merge branch 'airline-details' into new-reservations

ibr5500 committed:
Fix a duplicate button

ibr5500 committed:
Fix a linter error

ibr5500 committed:
Fix a dublicate button

TEMESGEN-G25 committed:
Delete BookFlight component

TEMESGEN-G25 committed:
Navigate to Reservation component after submitting

TEMESGEN-G25 committed:
Create reservations UI

TEMESGEN-G25 committed:
Render reservations from with airline details

TEMESGEN-G25 committed:
Merge branch 'dev' into airline-details

TEMESGEN-G25 committed:
Reinstall packages

TEMESGEN-G25 committed:
Merge pull request #6 from ibr5500/airline-details

Flight Reservation: Airline details

ibr5500 committed:
Functionality for delete

ibr5500 committed:
Remove unused code

ibr5500 committed:
Merge branch 'dev' into delete_flight

ibr5500 committed:
Update Home.js

ibr5500 committed:
Fix linter & routes errors

GOlivierNation committed:
Resolve conflicts

GOlivierNation committed:
Update Login.js

GOlivierNation committed:
Resolve conflict

GOlivierNation committed:
Resolve conflict

TEMESGEN-G25 committed:
Add an input element inside the reservationForm

TEMESGEN-G25 committed:
Merge branch 'new-reservations' of github.com:ibr5500/flight_reservation-frontend into new-reservations

TEMESGEN-G25 committed:
Fix typos

TEMESGEN-G25 committed:
Merge branch 'dev' into new-reservations

TEMESGEN-G25 committed:
Empty the airline form after submission

TEMESGEN-G25 committed:
Remove unused importations

TEMESGEN-G25 committed:
Render only the login page if loginToken is empty

TEMESGEN-G25 committed:
Navigate to the homepage after airline submission

TEMESGEN-G25 committed:
Create a handleSignOut function

TEMESGEN-G25 committed:
Render the login page if the loginToken is empty

TEMESGEN-G25 committed:
Merge pull request #7 from ibr5500/new-reservations

New reservations

TEMESGEN-G25 committed:
Merge branch 'new-reservations' into delete_flight

KanoCode committed:
Update styles for add flight

KanoCode committed:
Merge branch 'delete_flight' of https://github.com/ibr5500/flight_reservation-frontend into delete_flight

KanoCode committed:
update link path in Navigation panel

KanoCode committed:
Update styles for reservations list component

KanoCode committed:
Update reservation component

KanoCode committed:
update styles in deleted flights

KanoCode committed:
update routes

KanoCode committed:
update styles

KanoCode committed:
fix linter errors

TEMESGEN-G25 committed:
Add conditional redering if the user isn't log in

ibr5500 committed:
Fix some functionality

TEMESGEN-G25 committed:
Refresh the page whenever the page loads

TEMESGEN-G25 committed:
Fix linters

TEMESGEN-G25 committed:
Update README.md

KanoCode committed:
Add responsive table

KanoCode committed:
Import icons from react icons

KanoCode committed:
update styles

KanoCode committed:
Add back button

KanoCode committed:
add onclick handler to link

KanoCode committed:
fix linter errors

KanoCode committed:
Merge branch 'delete_flight' of https://github.com/ibr5500/flight_reservation-frontend into delete_flight

TEMESGEN-G25 committed:
Have proper reload after signing out

TEMESGEN-G25 committed:
Create snapshots for all tests

TEMESGEN-G25 committed:
Install necessary packages

TEMESGEN-G25 committed:
Create all tests

TEMESGEN-G25 committed:
Create redux tests file

TEMESGEN-G25 committed:
Create all necessary Mocks

TEMESGEN-G25 committed:
Fix linters

TEMESGEN-G25 committed:
Alert the page if the airline form is empty

TEMESGEN-G25 committed:
Make all your headers inside the table element

TEMESGEN-G25 committed:
Merge branch 'dev' into testing

TEMESGEN-G25 committed:
Merge pull request #9 from ibr5500/testing

Testing

ibr5500 committed:
Update README.md

ibr5500 committed:
Update README.md

ibr5500 committed:
Fix an error

ibr5500 committed:
Updates

ibr5500 committed:
Update README.md

TEMESGEN-G25 committed:
Add all project links

TEMESGEN-G25 committed:
Update README

mcrd25 commented on README.md:
```diff
@@ -0,0 +1,149 @@
+
+### Setup
+
+Clone the repository using the GitHub link provided below.
```
Should make clear set up for front-end vs back-end, maybe you can leave back-end set up out. But it should be clear how to run this app

mcrd25 commented on src/Components/AirlineForm.js:
```diff
@@ -0,0 +1,90 @@
+        <div className="mb-3">
+          <label htmlFor="airline-name" className="form-label">
+            Airline Name
+            <input
+              type="text"
+              className="form-control"
+              id="airline-name"
+              value={airline.name}
+              onChange={(e) => setAirline({ ...airline, name: e.target.value })}
+              placeholder="Name"
+            />
+          </label>
+        </div>
+        <div className="mb-3">
+          <label htmlFor="image" className="form-label">
+            Airline logo
+            <input
+              type="text"
+              className="form-control"
+              id="image"
+              value={airline.image}
+              onChange={(e) => setAirline({ ...airline, image: e.target.value })}
+              placeholder="Image"
+            />
+          </label>
+        </div>
+
+        <div className="mb-3">
+          <label htmlFor="price" className="form-label">
+            Price
+            <input
+              className="form-control"
+              type="text"
+              id="price"
+              value={airline.price}
+              onChange={(e) => setAirline({ ...airline, price: e.target.value })}
+              placeholder="Price"
+            />
+          </label>
```
[OPTIONAL] Since these are similar, it can be abstrcted futher to make code neater with loop

mcrd25 commented on src/Components/AirlineForm.js:
```diff
@@ -0,0 +1,90 @@
+            />
+          </label>
+        </div>
+        <div className="mb-3">
+          <label htmlFor="image" className="form-label">
+            Airline logo
+            <input
+              type="text"
+              className="form-control"
+              id="image"
+              value={airline.image}
+              onChange={(e) => setAirline({ ...airline, image: e.target.value })}
+              placeholder="Image"
+            />
+          </label>
+        </div>
```
[OPTIONAL] This is possibly a candidate for resuable componenets. These sets of elements repeat a coupel of times. Would make your code easier to read.

mcrd25 commented on src/Components/SignUp.js:
```diff
@@ -0,0 +1,109 @@
+                  id="username-input"
+                  value={newuser.username}
+                  onChange={(e) => setNewuser({ ...newuser, username: e.target.value })}
+                  placeholder="UserName"
+                />
+              </label>
+            </div>
+            <div className="">
+              <label htmlFor="email-input" className="w-100 my-2">
+                <input
+                  id="email-input"
+                  className="form-control py-2  px-4 "
+                  type="email"
+                  value={newuser.email}
+                  onChange={(e) => setNewuser({ ...newuser, email: e.target.value })}
+                  placeholder="Email"
+                />
+              </label>
+            </div>
+            <div className="">
+              <label htmlFor="password-input" className="w-100 my-2">
+                <input
+                  id="password-input"
+                  className="form-control py-2  px-4"
+                  type="password"
+                  value={newuser.password}
+                  onChange={(e) => setNewuser({ ...newuser, password: e.target.value })}
+                  placeholder="Password"
+                />
+              </label>
+            </div>
+            <div className="button-container w-50">
+
+              <button type="submit" className="btn sign-up my-2">
+                SignUp
+              </button>
+              <Link to="/" className="btn log-in">
+                Login
+              </Link>
+
```
[OPTIONAL] Is there a cleaner way to write this?

mcrd25 commented on src/Components/Home.js:
```diff
@@ -0,0 +1,38 @@
+      </div>
+      <div className="container">
+        <div className="flights">
+          {!(dataList) ? <p>There are no flight yet!!</p>
```
[OPTIONAL]
Is thereto delay in loading this until dataList for sure does not have data? I see there are no flights yet for a second or two before flights are loaded.

mcrd25 reviewed (changes requested):
Hi Team (@NatiG25 @GOlivierNation @ibr5500 @KanoCode ), 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions
- Currenlty, your home page with the list of flights does not follow the design from template (the slider) - Please address this
- There are some candidates to make some of these html components reusable.
- Consider making your code as cleaner

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

TEMESGEN-G25 committed:
Add extra design

ibr5500 committed:
Add some layouts for home page and some changes in login page

ibr5500 committed:
Fix Stylelint errors

ibr5500 committed:
Updates in the login component

ibr5500 committed:
Remove unused code

ibr5500 committed:
Fix linter error

TEMESGEN-G25 committed:
Upgrade snapshots & clean code

Meltrust reviewed (approved):
Hi @NatiG25,


## Wow, you did it 🎉
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### Unless you want to add more features, go ahead to your final presentation ▶️ ▶️ ▶️ 
### You are about to finish the Microverse program, you have come a long way!!!  Congratulations, and good luck in the software industry.  I'll see you there! 🎉 


### To highlight

- You made the suggested refactors✔️ 
- The app works✔️ 
- Tests are passing✔️
- Congratulations! 🏁 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

TEMESGEN-G25 commented:
@Meltrust & @mcrd25 Thank you both for all your suggestions :+1: :smile:.

![](https://media.giphy.com/media/W7DHaxkSE4Lpm/giphy.gif)

**Note**: All optional & required changes have been implemented.

TEMESGEN-G25 committed:
Update README.md

snyk-bot committed:
fix: upgrade react-icons from 4.4.0 to 4.6.0

Snyk has created this PR to upgrade react-icons from 4.4.0 to 4.6.0.

See this package in npm:
https://www.npmjs.com/package/react-icons

See this project in Snyk:
https://app.snyk.io/org/ibr5500/project/1e1233e6-c823-405d-aa75-ef5669affdee?utm_source=github&utm_medium=referral&page=upgrade-pr

TEMESGEN-G25 committed:
Update README.md

ibr5500 committed:
Merge pull request #15 from ibr5500/snyk-upgrade-3c4380c35435cf9e7379bd30a42ef31a

[Snyk] Upgrade react-icons from 4.4.0 to 4.6.0

ibr5500 committed:
Fix linter errors & update libraries

ibr5500 committed:
Update README.md

TEMESGEN-G25 committed:
Update README.md

TEMESGEN-G25 committed:
Change baseURL

TEMESGEN-G25 committed:
Fix linters

TEMESGEN-G25 committed:
Change reload time

TEMESGEN-G25 committed:
Change login reload time

TEMESGEN-G25 committed:
Delete unnecessary files

TEMESGEN-G25 committed:
Add a favicon

TEMESGEN-G25 committed:
Add more relaod time
